pilot/pkg/model/service.go

```diff
@@ -70,8 +71,13 @@ type Service struct {
 	Hostname host.Name `json:"hostname"`
 
 	// Address specifies the service IPv4 address of the load balancer
+	// Do not access directly. Use GetServiceAddressForProxy
```

Comment:
can we make it private? or will that break some stuff (not sure if we rely on json marshal for example)

Reply:
we can't. because the service registry code uses this variable directly (we could put a setter in there I guess). And there are a ton of tests across the code that use this variable. So any change would be a PR of its own